prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -137,15 +155,8 @@ object FhirPathUtils : Logging {
 
             evaluated[0] is TimeType -> convertTimeToHL7(evaluated[0] as TimeType)
 
-            else -> {
-                try {
-                    defaultPathEngine.convertToString(evaluated[0])
-                } catch (e: Exception) {
-                    val msg = "Could not parse ${evaluated[0]} to string."
-                    logger.error(msg, e)
-                    throw HL7ConversionException(msg, e)
-                }
-            }
+            // Use the string representation of the value for any other types.
+            else -> evaluated[0].toString()
```

Comment:
Just using toString threw errors for me in one of the random one off tests I did. I think its worth using `defaultPathEngine.convertToString` instead

Reply:
Changed.  I see that it actually calls `primitiveValue`